subsys/net/lib/download_client/src/download_client.c

```diff
@@ -622,6 +616,11 @@ int download_client_start(struct download_client *client, const char *file,
 
 	if (IS_ENABLED(CONFIG_COAP)) {
 		coap_block_init(client, from);
+		/* Set socket timeout, if configured */
```

Comment:
I think this is still wrong. You should not do anything for the socket because CONFIG_COAP is defined or not. It does not say anything about our use of TCP or UDP socket here.

If the flag is called `CONFIG_DOWNLOAD_CLIENT_UDP_SOCK_TIMEO_MS` the if should say `IPPROTO_UDP || IPPROTO_DTLS_1_2` or something.

Also the coap stuff is called in vain if we are not using coap

Reply:
Thank you, a fix for this issue is here: https://github.com/nrfconnect/sdk-nrf/pull/3431/commits